### PR TITLE
Update spring, httpclient, databind versions

### PR DESCRIPTION
**Pull request by jonesleonard**
Update Spring version to 4.3.17, Apache HTTP Client to 4.5.6
and Jackson databind to 2.8.11.1 in order to prevent any
potential security vulnerabilities

### CODE REVIEW

**Comment by jonesleonard** [comment]
First time making a pull request and contributing. Became interested when I saw mention of Hacktoberfest in the opensource channel on slack. 

Please let me know if I need to do anything else or modify anything. Saw these vulnerabilities show up in the scan and thought I could contribute.

File: samples/samples-springmvc/pom.xml [hunk 1]
```diff
@@ -9,8 +9,9 @@
     <version>0.0.1-SNAPSHOT</version>
     <packaging>war</packaging>
     <properties>
-        <spring.version>4.0.5.RELEASE</spring.version>
+        <spring.version>4.3.17.RELEASE</spring.version>
         <jackson.version>2.2.3</jackson.version>
+        <jackson.databind.version>2.8.11.1</jackson.databind.version>
```
Comment (pauljamescleary) [inline samples/samples-springmvc/pom.xml]: can we update jackson.version as well?  Just noticed there is no `2.8.11.1` but there is a `2.8.11`

**Comment by pauljamescleary** [comment]
@jonesleonard thanks for the contribution :)

Unfortunately, the build does not inspect the integrity of the sample project.  Were you able to run / test this locally?

File: samples/samples-springmvc/pom.xml (hunk 1 above)
Comment (pauljamescleary) [reply samples/samples-springmvc/pom.xml]: Better yet, is there any reason we can't go to 2.9.7?  Seems like both databind and core have that version.

Comment (jonesleonard) [reply samples/samples-springmvc/pom.xml]: Okay, I'll go ahead with 2.9.7 for both databind and core

**Comment by jonesleonard** [comment]
> @jonesleonard thanks for the contribution :)
> 
> Unfortunately, the build does not inspect the integrity of the sample project. Were you able to run / test this locally?

Just tried a `mvn compile` in the samples directory and received an error about this import statement: 

`import com.comcast.money.japi.TraceFriendlyExecutors;` 

I couldn't find a reference to the japi package or TraceFriendlyExecutors class in the project. Any suggestions? 

https://github.com/Comcast/money/blob/master/samples/samples-springmvc/src/main/java/com/comcast/money/samples/springmvc/config/AppConfig.java#L10

**Comment by pauljamescleary** [comment]
@jonesleonard I just noticed that this PR is coming off of the main repository.

Can you:
1. Create a fork of this main repository to your own account
2. Follow the github flow process to create your PR (redo what you have)
3. Close this PR
4. Open a new PR working off your fork.

Let me know if you have any questions.  That is generally the process for opensource.

**Comment by pauljamescleary** [comment]
Also, just noticed that this sample relies on spring 3.  We have more work to do in that regard.

What has to be done is:

1. Move all the files from the `money-spring3` sub module to `money-spring`.  Ensure the package names are `spring` instead of `spring3`.  For example, `SpringTracer` lives in `package com.comcast.money.spring3`, it would need to live in `com.comcast.money.spring`
2. Delete the `money-spring3` submodule
3. Update the sample project to use exclusively the new `money-spring` module.

For example, you can see in a file called `AppConfig` in the sample project the following code...

```java
import com.comcast.money.spring3.TracedMethodInterceptor;
import com.comcast.money.spring3.TracedMethodAdvisor;
import com.comcast.money.spring3.SpringTracer;

@Configuration
@ComponentScan(basePackages = {"com.comcast.money.samples.springmvc", "com.comcast.money.spring3"})
```

We would need to change `spring3` to just `spring`.

Let me know if you have questions.  Will be glad to walk you through this.

**Comment by kristakhare** [comment]
thanks for your help on this @jonesleonard and @paulcleary

**Pull request closed without merging**